### ISSUE TITLE
MsalGuard should acquire tokens during loginInteractively  

### ISSUE THREAD

**Issue by rpd10** [AzureAD/microsoft-authentication-library-for-js#1851]
## Library
- [ ] `msal@1.x.x` or `@azure/msal@1.x.x`
- [ ] `@azure/msal-browser@2.x.x`
- [ ] `@azure/msal-angular@0.x.x`
- [x] `@azure/msal-angular@1.x.x`
- [ ] `@azure/msal-angularjs@1.x.x`

## Description
MsalGuard does not currently acquire a token if you navigate to a protected route without previously being authenticated.  This leads to some inconsistent behavior regarding the `msal:` events that are dispatched.  I propose that loginInteractively should attempt to `acquireTokenSilent` if using the popup flow (using the redirect flow should not need any change).  

I would be willing to contribute a PR for this feature.

**Comment by jasonnutter** [comment]
@rpd10 Can you clarify what you mean? What type of tokens are you looking for, ID tokens or access tokens? MSAL Guard checks to see if you are logged in, and if not, sends you through the login flow (which will acquire an ID token).

**Comment by vajnorcan** [comment]
@rpd10 @jasonnutter 
I've had the same experience where I've setup in the **app.module.ts** the **MsalModule.forRoot()** and also the **MsalInterceptors**. Then I wanted to have one of my routes to be guarded with the **MsallGuard**. So typically, the user is not yet prompted to login until he hits the correct route.
The issue was, even when I was hitting the non-msal-guarded route,  I was getting the error from the msal library.

```
core.js:4002 ERROR Error: Uncaught (in promise): ClientAuthError: User login is required. For silent calls, request must contain either sid or login_hint
ClientAuthError: User login is required. For silent calls, request must contain either sid or login_hint  
```

and also the popup to prompt the user didn't show up once I've tried to hit the Msal guarded route.

But once I've removed the **MsalInterceptors** it was working as expected. 

So @rpd10 try to remove the interceptors.
@jasonnutter would be probably worth of either providing a guidance on this error or the correct implementation, or just let the interceptor fail gracefully.
